### ISSUE TITLE
(Feature request) Export Notes option to include internal links

### ISSUE THREAD

**Issue by deajan** [zadam/trilium#3399]
### Describe feature

When exporting notes, one can choose to export current note or current note with all its descendants.
While this works well in hierarchy organized notes, it won't work with internal links.

It would be a nice to have function to export notes with all it's descendants and internal links, so the exported note would be complete, regardless of it's structure in Trilium.

**Comment by zadam** [comment]
Hi, I'm not sure I understand. When you export a subtree into ZIP, the internal links (between notes within the exported subtree) are preserved. You can open the individual HTML files generated by the export and click through the links.

**Comment by deajan** [comment]
Internal links are preserved only when the internal link points to a note from the subtree.

Example:

```
root
 |
 |___NOTE 1
 |       |_______SUBNOTE 1
 |___NOTE 2
         |_______SUBNOTE 2
```

If NOTE 1 has two internal links to both SUBNOTE 1 and SUBNOTE 2, only SUBNOTE 1 is exported, leaving NOTE 1 incomplete.

**Comment by deajan** [comment]
> Yes, but I don't see how we could preserve links to notes which are not part of the export? They would be dead links anyway (since those target notes are not in the export). What purpose would that serve?

Indeed, as of today, my exports have dead links to notes outside of the export root.
The point is not "preserve links", but rather include the linked notes in the export schema.

Here's a less conceptual example of what I am doing:

I have a knowledge tree for Linux (RHEL, Debian) and generic instruction notes that apply for every linux flavor (ie mysql or postfix config notes).

```
root
|______LINUX
           |_______RHEL
           |            |_______LAMP stack (includes links to MySQL and Postfix)
           |            |_______ERP Server (includes links to Redis)
           |________DEBIAN
           |            |_______LAMP stack (includes links to MySQL and Postfix)
           |________Configurations
                         |_________MySQL
                         |_________Postfix
                         |_________Redis

When I export a RHEL documentation, I'll have:
- RHEL note
- RHEL/LAMP stack note
- RHEL/ERP Server note

Links between those 3 notes work.

Now the LAMP and ERP internal links to the configurations will be dead links.
Since those links apply not only to RHEL, it's obvious to have them stored outside the RHEL tree.

It would be wonderful to have those linked notes exported too, regardless of their storage place, as long as the'yre linked to.

My reasoning behind this is that Trilium allows to structure notes in multiple ways to adapt to the cognitive process of the author.  So exports should be complete regardless of the structure of the notes.

**Comment by zadam** [comment]
@deajan thanks, I now see what you mean.

**Comment by eliandoran** [comment]
@deajan , is this request still relevant to you?
I wonder if the complexity of handling complex closure/cycles for export is worth it, to be honest. 🤔

**Comment by deajan** [comment]
TBH, it would be a nice to have in order to produce "complete" exports without wondering what could fail at the end.
I have no idea what code complexity it creates, but it's not vital for me. Perhaps let this live on a wish list of "the perfect Trilium experience(TM)" ?
